sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/SimpleHttpHeartbeatSender.java

```diff
@@ -69,7 +70,7 @@ public boolean sendHeartbeat() throws Exception {
         }
 
         InetSocketAddress addr = new InetSocketAddress(addrInfo.r1, addrInfo.r2);
-        SimpleHttpRequest request = new SimpleHttpRequest(addr, TransportConfig.getHeartbeatApiPath());
```

Comment:
If a new type was introduced it's highly recommended to pass it into query implementations directly.

Reply:
I'm sorry, I didn't get your point here

Reply:
> If a new type was introduced it's highly recommended to pass it into query implementations directly.

I'm sorry, I didn't get your point here. Could you give me a hint?

Reply:
> > If a new type was introduced it's highly recommended to pass it into query implementations directly.
> 
> I'm sorry, I didn't get your point here. Could you give me a hint?

It's related to the previous design (Tuple3/class Endpoint{}). If a new class was introduced there's no need to split it into three separated arguments I think.

Reply:
> > > If a new type was introduced it's highly recommended to pass it into query implementations directly.
> > 
> > 
> > I'm sorry, I didn't get your point here. Could you give me a hint?
> 
> It's related to the previous design (Tuple3/class Endpoint{}). If a new class was introduced there's no need to split it into three separated arguments I think.

After using Endpoint, I will modify it here and directly pass in Endpoint in the constructor of SimpleHttpRequest to replace protocol and socketAddress

Reply:
> 
> After using Endpoint, I will modify it here and directly pass in Endpoint in the constructor of SimpleHttpRequest to replace protocol and socketAddress

Yes, that's it. 
Meanwhile, it won't be a backward compatible change and we don't have the perfect one exactly.